awx/api/serializers.py

```diff
@@ -1241,6 +1241,19 @@ def get_summary_fields(self, obj):
         return ret
 
 
+class DirectTokenSerializer():
```

Comment:
I don't think this is something we'd ever expose to the API.  Given that you're just using `oauthlib.common.generate_token` and calling ORM code, why not just do _that_ instead in `tasks.py` ?

Reply:
I thought about that initially but was thinking maybe this could be reused someday. How would you feel about having a get_token method in the User object? Then you could do something like: 
env['TOWER_OAUTH_TOKEN'] = User.objects.get(id=project_update.created_by_id).get_token('<description>')

Reply:
In the end I put this into the models/oauth as the User object is from django, not AWX.